File: frontend/src/api/infrastructure.ts
import apiClient from './client';
import type { BackupServer, ProxyServer, Repository } from '../types';

export const getBackupServers = () =>
  apiClient.get<BackupServer[]>('/infrastructure/backup-servers').then((r) => r.data);

export const getProxyServers = () =>
  apiClient.get<ProxyServer[]>('/infrastructure/proxy-servers').then((r) => r.data);

export const getRepositories = () =>
  apiClient.get<Repository[]>('/infrastructure/repositories').then((r) => r.data);

export const getGrafanaToken = () =>
  apiClient.get<{ url: string; token: string }>('/infrastructure/grafana-token').then((r) => r.data);

File: backend/app/api/infrastructure.py
from fastapi import APIRouter
from app.collectors import get_collector
from app.core.config import settings

router = APIRouter(prefix="/infrastructure", tags=["infrastructure"])


@router.get("/backup-servers")
def backup_servers():
    return get_collector().get_backup_servers()


@router.get("/proxy-servers")
def proxy_servers():
    return get_collector().get_proxy_servers()


@router.get("/repositories")
def repositories():
    return get_collector().get_repositories()


@router.get("/grafana-token")
def grafana_token():
    return {"url": settings.GRAFANA_URL, "token": settings.GRAFANA_API_KEY}
